core/shared/src/main/scala/org/http4s/h2/H2Keys.scala

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2013 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s.h2
+
+import cats.effect._
+import org.typelevel.vault._
+
+object H2Keys {
+
+  /** Client Side Key To Try Http2-Prior-Knowledge
+    * which means immediately using http2 without any upgrade mechanism
+    * but is invalid if the receiving server does not support the
+    * mechanism.
+    */
+  val Http2PriorKnowledge: Key[Unit] = Key.newKey[SyncIO, Unit].unsafeRunSync()
```

Comment:
Are you going to propagate this to other clients than Ember? 
Also, I remember some folks have strong opinions on the usage of `lazy val`s, but it seems reasonable here since this value won't be used across the 'core' module itself.

Reply:
what I want is this:

https://github.com/davenverse/http4s-grpc/pull/89

But @armanbilge did not want to add the suggested middleware.

Reply:
@armanbilge I expressed a naive desire to move this library into the `http4s` ecosystem sometime ago. Any changes to the standpoint?

Reply:
- https://github.com/http4s/http4s/discussions/7017

No objections. I think we should move it :)

Reply:
so now what. Should I continue in the new repo?

Reply:
The location of that repo doesn't affect this PR though :)